community/cypher/cypher-compiler-2.3/src/main/scala/org/neo4j/cypher/internal/compiler/v2_3/commands/expressions/ToIntFunction.scala

```diff
@@ -37,16 +37,12 @@ case class ToIntFunction(a: Expression) extends NullInNullOutExpression(a) {
       v.longValue()
     case v: String =>
       try {
-        v.toLong
+        val d = BigDecimal(v)
+        if (d <= Long.MaxValue && d >= Long.MinValue) d.toLong
+        else throw new ParameterWrongTypeException(s"$v cannot be represented as an 8-byte integer")
```

Comment:
Probably just use the same "integer is too large" message, rather than mentioning byte numbers?